### ISSUE TITLE
[Bug]: If you run PackageSearchResource with parameter `null` as filter it will wait forever

### ISSUE THREAD

**Issue by arthurvb** [NuGet/Home#12178]
### NuGet Product Used

NuGet SDK

### Product Version

nuget.exe

### Repro Steps & Context

Run the following code
```
var repo = Repository.Factory.GetCoreV2(source);
var packageSearchResource = await repo.GetResourceAsync<PackageSearchResource>();
CancellationToken cancellationTokenSearch = new CancellationTokenSource(TimeSpan.FromSeconds(5.0)).Token;
var results = await packageSearchResource.SearchAsync("MyPackage", 
                        null, 0, 1, NullLogger.Instance,
                        cancellationTokenSearch);
```
it will run forever.

However the following code will execute correctly:
```
var repo = Repository.Factory.GetCoreV2(source);
var packageSearchResource = await repo.GetResourceAsync<PackageSearchResource>();
CancellationToken cancellationTokenSearch = new CancellationTokenSource(TimeSpan.FromSeconds(5.0)).Token;
var results = await packageSearchResource.SearchAsync("MyPackage", 
                        new SearchFilter(false), 0, 1, NullLogger.Instance,
                        cancellationTokenSearch);
```

Even though the first code is false I would expect an error instead of running forever.

Thank you for your time and effort!

**Comment by nkolev92** [comment]
Hey @arthurvb 
Thanks for filing the issue. Unfortunately I can't repro.

In the first case, I'm getting an NRE, 

```cs
 NuGet.Protocol.Tests.PackageSearchResourceV2FeedTests.Test
   Source: PackageSearchResourceV2FeedTests.cs line 22
   Duration: 492 ms

  Message: 
System.NullReferenceException : Object reference not set to an instance of an object.

  Stack Trace: 
V2FeedQueryBuilder.BuildSearchUri(String searchTerm, SearchFilter filters, Int32 skip, Int32 take) line 65
V2FeedParser.Search(String searchTerm, SearchFilter filters, Int32 skip, Int32 take, ILogger log, CancellationToken token) line 253
PackageSearchResourceV2Feed.SearchAsync(String searchTerm, SearchFilter filters, Int32 skip, Int32 take, ILogger log, CancellationToken cancellationToken) line 44
PackageSearchResourceV2FeedTests.Test() line 27
--- End of stack trace from previous location where exception was thrown ---
```

My test case is 
```

        [Fact]
        public async Task Test()
        {
            var repo = Repository.Factory.GetCoreV2(new Configuration.PackageSource("https://www.nuget.org/api/v2"));
            var packageSearchResource = await repo.GetResourceAsync<PackageSearchResource>();
            CancellationToken cancellationTokenSearch = new CancellationTokenSource(TimeSpan.FromSeconds(5.0)).Token;
            var results = await packageSearchResource.SearchAsync("MyPackage",
                                    null, 0, 1, NullLogger.Instance,
                                    cancellationTokenSearch);
        }
```


What kind of feed are you using?

```cs
            var repo = Repository.Factory.GetCoreV2(new Configuration.PackageSource("C:\\Users\\Nikolche\\.nuget\\packages"));
            var repo = Repository.Factory.GetCoreV2(new Configuration.PackageSource("https://api.nuget.org/v3/index.json"));
            var repo = Repository.Factory.GetCoreV2(new Configuration.PackageSource("https://www.nuget.org/api/v2"));
```

All of the 3 tests had an NRE.

**Comment by arthurvb** [comment]
Dear @nkolev92,

Thank you for your quick and thorough reply. Upon re-inspection, I believe the issue may have been caused by trying to call the methods from a WPF app with `.Result`, which can result in a deadlock. When I made the method asynchronous, I forgot to add error-handling, so the breakpoint after the method was not hit and I incorrectly assumed that the method was still executing.

I hope other people can learn from this experience.

Once again, thank you for your quick and kind response.
